packages/realm-web/src/transports/AuthenticatedTransport.ts

```diff
@@ -59,19 +60,37 @@ export class AuthenticatedTransport implements Transport {
      * @param request The request to issue towards the server
      * @param user The user used when fetching, defaults to the `app.currentUser`.
      *             If `null`, the fetch will be unauthenticated.
+     * @param retries How many times was this request retried?
      * @returns A response from requesting with authentication.
      */
-    public fetch<RequestBody extends any, ResponseBody extends any>(
+    public async fetch<RequestBody extends any, ResponseBody extends any>(
         request: Request<RequestBody>,
-        user: Realm.User | null = this.userContext.currentUser,
+        user: User | null = this.userContext.currentUser,
+        retries = 0,
     ): Promise<ResponseBody> {
-        return this.transport.fetch({
-            ...request,
-            headers: {
-                ...this.buildAuthorizationHeader(user),
-                ...request.headers,
-            },
-        });
+        try {
+            // Awaiting to intercept errors being thrown
+            return await this.transport.fetch({
+                ...request,
+                headers: {
+                    ...this.buildAuthorizationHeader(user),
+                    ...request.headers,
+                },
+            });
+        } catch (err) {
+            if (
+                user &&
+                retries === 0 &&
+                err instanceof MongoDBRealmError &&
+                err.statusCode === 401
```

Comment:
Could there be a case where the endpoint returns `403` (forbidden) from an outdated token - if a user has been granted new/extended access, but the client has not updated its access token yet?
(I guess this depends on how much of the access rights are stored in the JWT).

Reply:
I don't think so ... it seems to me only 401 are returned when a refresh of the access token would resolve the issue.